### PR TITLE
fixed: cpu usage is high on numeric gc

### PR DESCRIPTION
**Pull request by MeirShpilraien**
Solution: run smaller gc iterations and release the lock between iterations.

### CODE REVIEW

**Comment by MeirShpilraien** [comment]
@mnunberg I am not sure, some of the checks performed are only related to the gc (the spec_unique_id and the revisionId). So the refresh function would contains only 3 lines ...

File: src/gc.c [hunk 1]
```diff
@@ -61,11 +61,7 @@ GarbageCollectorCtx *NewGarbageCollector(const RedisModuleString *k, float initi
   GarbageCollectorCtx *gc = malloc(sizeof(*gc));
 
   *gc = (GarbageCollectorCtx){
-      .timer = NULL,
-      .hz = initialHZ,
-      .keyName = k,
-      .stats = {},
-      .rdbPossiblyLoading = 1,
+      .timer = NULL, .hz = initialHZ, .keyName = k, .stats = {}, .rdbPossiblyLoading = 1,
```
Comment (itamarhaber) [inline src/gc.c]: Formatting: you just changed that to multiple newlines in the previous commit, no? Personally, I prefer multiline.

File: src/gc.c [hunk 2]
```diff
@@ -267,18 +255,39 @@ size_t gc_NumericIndex(RedisModuleCtx *ctx, GarbageCollectorCtx *gc, int *status
   }
 
   NumericRangeNode *nextNode = NextGcNode(numericGcCtx);
-  InvertedIndex_Repair(nextNode->range->entries, &sctx->spec->docs, 0,
-                       nextNode->range->entries->size, &bytesCollected, &recordsRemoved);
 
-  gc_updateStats(sctx, gc, recordsRemoved, bytesCollected);
+  int blockNum = 0;
+  int num = 100;
```
Comment (itamarhaber) [inline src/gc.c]: I think this should be a default define (and maaaybe a config option? later)

File: src/gc.c (hunk 1 above)
Comment (mnunberg) [reply src/gc.c]: @itamarhaber we're trying to use a formatter file for universal formatting - i don't like auto-formatters, but it's the easiest to work with.

Comment (MeirShpilraien) [reply src/gc.c]: just installed a clang plugin on the eclipse so it should be fine from now on

**Pull request merged**